ptero_workflow/api/v1/api.py

```diff
@@ -12,6 +12,14 @@
 api.add_resource(views.WorkflowDetailView,
     '/workflows/<int:workflow_id>', endpoint='workflow-detail')
 
+api.add_resource(views.ExecutionInputsView,
+    '/executions/<int:execution_id>/inputs',
+    endpoint='execution-inputs')
+
+api.add_resource(views.ExecutionOutputsView,
```

Comment:
What's the advantage of immutable endpoints?  I mean either way you're modifying the information content of the entity we call an execution, regardless of the endpoints.